### PR TITLE
Transactions for tests!

### PR DESCRIPTION
**Pull request by tschuy**
This makes things fast!

```
  209 passing (6s)
```

### CODE REVIEW

**Comment by tschuy** [comment]
Ok, this is ready for review. 

This removes the ability to use sqlite as a database backend, which, given some of the bugs I found, isn't a bad idea.

To run tests, you'll need to make sure your ``PG_CONNECTION_STRING`` connects to an empty database that has had migrations run. This is the same string as the dev uses, so I'd switch between database names before running tests!

**Comment by Kennric** [comment]
Um, that's discomfiting - can the test runner set that variable? That looks like a recipe for losing data otherwise

**Comment by tschuy** [comment]
Ok, I've made a few changes:

1. now it supports both sqlite and postgres backends. to run sqlite tests, run ``npm test``; to run postgres tests, run ``npm run test_pg``.

2. it allows you to specify a separate database connection for tests -- ``TEST_PG_CONNECTION_STRING`` is what is used now.

should be good to go now?

File: knexfile.js [hunk 1]
```diff
@@ -6,13 +6,23 @@ module.exports = {
...
+  development_pg: {
+    client: 'pg',
+    connection: process.env.PG_CONNECTION_STRING
+  },
+
+  mocha_sqlite: {
     client: 'sqlite3',
     connection: {
-      filename: ':memory:'
+      filename: './test.sqlite3'
```
Comment (Kennric) [inline knexfile.js]: Why not in memory?

Comment (tschuy) [inline knexfile.js]: ah, oops. my mistake.

**Comment by LyonesGamer** [comment]
+1

**Pull request merged**